component/client/core/ClientChat.tsx

```diff
@@ -28,7 +29,17 @@ class ClientChat extends React.Component<IClientChatProps, IClientChatState> {
     this.messenger = <Messenger handleSend={this.handleSend}/>;
   }
 
+  public componentDidMount() {
```

Comment:
I don't feel this is necessary, whenever the URL changed, a new RoomService should be created. It is not expensive to create one.